src/pages/components/HistoryItem.qml

```diff
@@ -0,0 +1,106 @@
+import QtQuick 2.6
+import Sailfish.Silica 1.0
+import Sailfish.Browser 1.0
+
+ListItem {
+    id: root
+
+    property string search
+
+    function remove() {
+        view.model.remove(model.index)
+    }
+
+    function openLink() {
+        webView.load(model.url, model.title)
+        pageStack.pop()
+    }
+
+    width: view.width
+    menu: contextMenuComponent
+
+    ListView.onAdd: AddAnimation { target: root }
+
+    Column {
+        anchors {
+            left: parent.left
+            leftMargin: Theme.horizontalPageMargin
+            right: rightColumn.left
+            rightMargin: Theme.paddingSmall
+        }
+        HistoryLabel {
+            id: titleText
+            text: Theme.highlightText(model.title, search, Theme.highlightColor)
+        }
+
+        HistoryLabel {
+            text: Theme.highlightText(model.url, search, Theme.highlightColor)
+            opacity: Theme.opacityHigh
+        }
+    }
+    Column {
+        id: rightColumn
+        width: dateLabel.implicitWidth
+        anchors {
+            right: deleteButton.left
+            rightMargin: Theme.paddingSmall
+        }
+        HistoryLabel {
+            id: dateLabel
+            text: Format.formatDate(model.date, Formatter.DateMediumWithoutYear)
```

Comment:
This is not at design. Only time and time should be following platform current clock.

Section header will depicts relative time view => "Today", "Yesterday", "Last week", and so on.